scipy/stats/tests/test_stats.py

```diff
@@ -4178,6 +4178,18 @@ def test_nan_policy(self):
         assert_almost_equal(stats.kruskal(x, x, nan_policy='omit'), (0.0, 1.0))
         assert_raises(ValueError, stats.kruskal, x, x, nan_policy='raise')
         assert_raises(ValueError, stats.kruskal, x, x, nan_policy='foobar')
+        
+    def test_large_no_samples(self):
+        # Test to see if large samples are handled correctly.
+        n = 50000
+        np.random.seed(0)  # Make the test reproducible.
+        x = np.random.randn(n)
+        y = np.random.randn(n) + 50
+        h, p = stats.kruskal(x, y)
+        expected_p = 0
+        assert_approx_equal(p, expected_p)
+        expected_h = 74999.250007499941
```

Comment:
Can you please add a comment explaining where this number came from and on what tolerance level one expects an approximate equality. E.g., if I increase `n`, how does the expected value change?

Reply:
> The test would also be broken if numpy changes the way in which random numbers are generated in the future.

This won't happen, numpy gives a hard backwards compatibility guarantee on this. (one of the reasons, everyone's test suites will break otherwise)

Reply:
Testing ``p`` not ``h`` sounds okay though.